tools/clang/test/HLSLFileCheck/hlsl/control_flow/loops/struct_exit_exit_value.hlsl

```diff
@@ -50,8 +56,10 @@ float main(uint a : A, uint b : B, uint c : C) : SV_Target {
           return 1;
         }
 
-        array[(idx + i) % 5] += a;
+        uav1[i + offset] += a;
+        //array[(idx + i) % 5] += a;
```

Comment:
Since you are using `uav1` now I think you can get rid of `array` alltogether (the only remaining use at the return is optimized away anyway). The same goes for `idx`.
(Note that `array` isn't the only thing causing cleanup code behind the loop when lifetime intrinsics are generated: `ret` is already enough for this since it receives lifetime markers before being promoted to registers.)